WordPress/Classes/ViewRelated/Post/AbstractPostListViewController.swift

```diff
@@ -0,0 +1,1047 @@
+        }
+        
+        return index
+    }
+
+    func indexForFilterWithType(filterType: PostListStatusFilter) -> Int {
+        if let index = availablePostListFilters().indexOf({ (filter: PostListFilter) -> Bool in
+            return filter.filterType == filterType
+        }) {
+            return index
+        } else {
+            return NSNotFound
+        }
+    }
+    
+    func keyForCurrentListStatusFilter() -> String {
+        assert(false, "You should implement this method in the subclass")
+    }
+    
+    func currentFilterIndex() -> Int {
+        
+        let userDefaults = NSUserDefaults.standardUserDefaults()
+        
+        if let filter = userDefaults.objectForKey(keyForCurrentListStatusFilter()) as? Int
+            where filter < availablePostListFilters().count {
+            
+            return filter
+        } else {
+            return 0 // first item is the default
+        }
+     }
+    
+    func setCurrentFilterIndex(newIndex: Int) {
+        let index = currentFilterIndex()
+        
+        guard newIndex != index else {
+            return
+        }
+        
+        WPAnalytics.track(.PostListStatusFilterChanged, withProperties: propertiesForAnalytics())
+        NSUserDefaults.standardUserDefaults().setObject(newIndex, forKey: keyForCurrentListStatusFilter())
+        NSUserDefaults.resetStandardUserDefaults()
+        
+        recentlyTrashedPostObjectIDs.removeAll()
+        updateFilterTitle()
+        resetTableViewContentOffset()
+        updateAndPerformFetchRequestRefreshingCachedRowHeights()
+    }
+    
+    func updateFilterTitle() {
+        filterButton.setAttributedTitleForTitle(currentPostListFilter().title)
+    }
+    
+    func displayFilters() {
+        let titles = availablePostListFilters().map { (filter: PostListFilter) -> String in
+            return filter.title!
+        }
+        
+        let dict = [SettingsSelectionDefaultValueKey: availablePostListFilters()[0],
+                    SettingsSelectionTitleKey: NSLocalizedString("Filters", comment: "Title of the list of post status filters."),
+                    SettingsSelectionTitlesKey: titles,
+                    SettingsSelectionValuesKey: availablePostListFilters(),
+                    SettingsSelectionCurrentValueKey: currentPostListFilter()]
+        
+        let controller = SettingsSelectionViewController(style: .Plain, andDictionary: dict)
+        controller.onItemSelected = { [weak self] (selectedValue: AnyObject!) -> () in
+            if let strongSelf = self,
+                let index = strongSelf.availablePostListFilters().indexOf(selectedValue as! PostListFilter) {
+                
+                strongSelf.setCurrentFilterIndex(index)
+                strongSelf.dismissViewControllerAnimated(true, completion: nil)
+            }
+        }
+        
+        let navController = UINavigationController(rootViewController: controller)
+        
+        if UIDevice.isPad() {
```

Comment:
Agreed, but again, this PR is not the right place to change the logic of the underlying code.